src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1132,38 +1140,30 @@ private SolutionCompilationState ComputeFrozenSnapshot(CancellationToken cancell
 
         return newCompilationState;
 
-        void CheckDocumentStates<TDocumentState>(
+        static void AddMissingOrChangedFilePathMappings<TDocumentState>(
+            FilePathToDocumentIdsMap.Builder filePathToDocumentIdsMapBuilder,
             TextDocumentStates<TDocumentState> oldStates,
             TextDocumentStates<TDocumentState> newStates) where TDocumentState : TextDocumentState
         {
             if (oldStates.Equals(newStates))
```

Comment:
>Equals [](http://example.com/codeflow?start=26&length=6)

Is it worth doing doing the Equals check? In the case where they are the same, isn't it doing essentially the same work that the foreach does below?

Reply:
i def think it's worth it as it's a simple o(1) chekc that avoids a linear walk.